### PR TITLE
Generate mandatory XLS forms programmatically

### PR DESCRIPTION
**Pull request by Sujanadh**
## Summary

This pull request introduces a new feature to generate mandatory XLS forms directly within the codebase. It modularizes the workflow, making it easier to manage and scale form generation tasks.

## Features

Added a modular structure to generate XLS forms:

- `mandatory_fields.py:` Defines essential fields for all forms.
- `digitisation_fields.py:` Includes specific fields for digitization and verification processes.
- `choice_fields.py:` Handles choices required for mandatory fields.

Combined all components into a single pipeline to generate the final survey XLS form.

## Key Improvements
Added `submission_id` caculation field to update entity property with latest `submission_id`.
Created a function `standardize_xlsform_sheets` which:
- checks for the columns, converts them to lower case
- removes any empty rows, and 
- mainly checks the language labels

and creates a standard language label for any kind of label present in the user-defined xls form consequently making a standard xls form.

### CODE REVIEW

**Comment by spwoodcock** [comment]
> ## Update:
> 
> Actually, there was a way to have multiple `submission_ids` in a single entity just by concatenating it with a previous `submission_id`.
> 
>     * updated the caculations of `submission_ids` to append previous `submission_ids` as well.

That's huge!!

We can easily have a disambiguation dialog to select the submission on the frontend then 🎉 

💯 the workflow to choose - thanks for sticking to your guns and convincing me 😄

File: osm_fieldwork/form_components/choice_fields.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+]
+
+choices_df = pd.DataFrame(choices_data)
+
+digitisation_choices = [
+    {
+        "list_name": "digitisation_problem",
+        "name": "lumped",
+        "label::english(en)": "Lumped - one polygon (more than one building digitized as one)",
+        "label::swahili(sw)": "Lumped - poligoni moja (zaidi ya jengo moja limewekwa dijiti kuwa moja)",
```
Comment (spwoodcock) [inline osm_fieldwork/form_components/choice_fields.py:47]: This PR is perfect for now, but we should definitely consider moving all these translations out into a separate YAML or something, at some point.

This way users could update the translations via weblate or transifex (we will add this for UI translations soon anyway)

**Review by spwoodcock — APPROVED** [review]

File: osm_fieldwork/update_xlsform.py [hunk 2]
```diff
@@ -18,7 +21,48 @@
...
 FEATURE_COLUMN = "feature"
 NAME_COLUMN = "name"
+TYPE_COLUMN = "type"
 SURVEY_GROUP_NAME = "survey_questions"
+DEFAULT_LANGUAGES = {
+    "english": "en",
+    "french": "fr",
+    "spanish": "es",
+    "swahili": "sw",
+    "nepali": "ne",
```
Comment (spwoodcock) [inline osm_fieldwork/update_xlsform.py:31]: 🙌😁

If you can translate the other columns into Nepali then feel free to add them! (might help Tokha I imagine)

**Comment by spwoodcock** [comment]
Merge whenever 👍

You can make and push a new release too

```bash
cz bump --check-consistency
git push
git push --tag

#then release on github
```

File: osm_fieldwork/form_components/choice_fields.py (hunk 1 above)
Comment (Sujanadh) [reply osm_fieldwork/form_components/choice_fields.py:47]: Yeah I also thought about that :+1:

**Comment by Sujanadh** [comment]
I will get back to it sooner.

**Pull request merged**